logging.ts

```diff
@@ -0,0 +1,68 @@
+
+import { join } from "https://deno.land/std@0.223.0/path/mod.ts";
+interface LoggingConfig {
+    logDir: string;
+    infoLog: string;
+    errorLog: string;
+}
+
+interface Config {
+    logging: LoggingConfig;
+}
+
+const config: Config = JSON.parse(await Deno.readTextFile('config.json'));
+const infoPath = join(config.logging.logDir, config.logging.infoLog);
+const errorPath = join(config.logging.logDir, config.logging.errorLog);
+
+// interface for a access log
+export interface AccessLog {
+    method: string,
+    url: string,
+    protocol: string,
+    status: number,
+    size: number,
+    referer: string,
+    userAgent: string,
+    responseTime: number,
+    upstream_response_time: number,
+    backend: string,
+}
+
+// basic logging function
+export async function log(logType: string, message: string, context: object = {}) {
+    const entry = {
+        timestamp: new Date().toISOString(),
+        logType,
+        message,
+        ...context,
+    };
+
+    // if the logs directory does not exist, create it
+    try {
+        await Deno.mkdir(config.logging.logDir);
+    } catch (error) {
+        if (!(error instanceof Deno.errors.AlreadyExists)) {
+            console.error(`Failed to create log directory: ${error}`);
+        }
+    }
```

Comment:
You can just use the [`recursive` option](https://deno.land/api@v1.43.3?s=Deno.MkdirOptions#prop_recursive).

Reply:
Wait true